core/src/main/scala/chisel3/package.scala

```diff
@@ -210,29 +212,140 @@ package object chisel3 {
   implicit class PrintableHelper(val sc: StringContext) extends AnyVal {
 
     /** Custom string interpolator for generating Printables: p"..."
-      * Will call .toString on any non-Printable arguments (mimicking s"...")
+      * mimicks s"..." for non-Printable data)
       */
     def p(args: Any*): Printable = {
-      sc.checkLengths(args) // Enforce sc.parts.size == pargs.size + 1
-      val pargs: Seq[Option[Printable]] = args.map {
-        case p: Printable => Some(p)
-        case d: Data      => Some(d.toPrintable)
-        case any =>
-          for {
-            v <- Option(any) // Handle null inputs
-            str = v.toString
-            if !str.isEmpty // Handle empty Strings
-          } yield PString(str)
+      // P interpolator does not treat % differently - hence need to add % before sending to cf.
+      val t = sc.parts.map(_.replaceAll("%", "%%"))
+      StringContext(t: _*).cf(args: _*)
+    }
+
+    /** Custom string interpolator for generating formatted Printables : cf"..."
+      *
+      * Enhanced version of scala's `f` interpolator
+      * Each expression (argument) referenced within the string is
+      * converted to a particular Printable depending
+      * on the format specifier and type.
+      *
+      * For Chisel types referenced within the String
+      * %n - Returns Name Printable
```

Comment:
not sure how this list gets formatted, I think scaladoc generally uses wiki syntax, so maybe need some bullets and `<code>` blocks?
https://en.wikipedia.org/wiki/Help:Wikitext#Lists

Reply:
@mwachs5  tried using the bullet scheme with ** - still no effect - as in just regular ** gets printed. Do we have some examples within any of our source code trying to render bullets?

Reply:
I would think you'd want `*` single or `-`? Since the first `*` is for just the scaladoc itself? I'll look to see if we have something...

Reply:
Both `*` and `-` did not work. Specifically this is what I did 
```
* ==== For Chisel types referenced within the String ====
      * - %n - Returns [[Name]] Printable.
      * - %N - Returns [[FullName]] Printable.
      * -  %b,%d,%x,%c = Only applicable for types of [[Bits]] or dreived from it. - returns ([[Binary]],[[Decimal]],
      * [[Hexadecimal]],[[Character]]) Printable respectively.
      * Default if no specifier given is to call [[Data.toPrintable]] on the Chisel Type.
```

Reply:
the - gets printed instead of bullet.

Reply:
The first * on the left is part of the scaladoc.

Reply:
for the lists, try adding newlines around them and then using `-`:

```
The markup for list blocks looks like:

/** Here is an unordered list:
  * 
  *   - First item
  *   - Second item
  *     - Sub-item to the second
  *     - Another sub-item
  *   - Third item
  *
  * Here is an ordered list:
  *
  *   1. First numbered item
  *   1. Second numbered item
  *     i. Sub-item to the second
  *     i. Another sub-item
  *   1. Third item
  */
```

Reply:
@mwachs5  worked!! Yaayyy!! Thanks!!! - Should have reached out earlier - spent ~ 20 mins or so on this - but in the process got enlightened a bit about ScalaDocs too!

Reply:
What was missing was the "extra space" before the -